<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Rajamani Nirmalaruban" />

<meta name="date" content="2025-02-20" />

<title>Introduction of ‘geneNR’</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction of ‘geneNR’</h1>
<h4 class="author">Rajamani Nirmalaruban</h4>
<h4 class="date">2025-02-20</h4>



<div id="r-markdown" class="section level2">
<h2>R Markdown</h2>
<p><span style="font-size: 18px;font-weight: bold;">Introduction:</span></p>
<p><span style="font-size: 16px;">The geneNR package is designed to
streamline the post-GWAS (Genome-Wide Association Studies) analysis by
automating the identification of candidate genes within a user-defined
search window based on the identified SNPs (Single Nucleotide
Polymorphisms). This package provides robust support for candidate gene
analysis specifically for wheat and rice, making it an invaluable tool
for researchers working in the field of genomics.</span></p>
<p><span style="font-size: 18px;font-weight: bold;">Key
Features:</span></p>
<p><span style="font-size: 16px;"><strong><em>Automated Search and
Retrieval:</em></strong> geneNR simplifies the labor-intensive process
of manually searching and retrieving candidate genes. By leveraging the
package, researchers can save time and focus on interpreting results
rather than conducting repetitive search.</span></p>
<p><span style="font-size: 16px;"><strong><em>User-Defined Search
Window:</em></strong> The package allows users to define their own
search window parameters, providing flexibility and customization to
suit various research needs.</span></p>
<p><span style="font-size: 16px;"><strong><em>Support for Wheat and
Rice:</em></strong> With a focus on two of the most important staple
crops, wheat and rice, geneNR ensures targeted and relevant candidate
gene analysis, aiding in crop improvement and genetic research
efforts.</span></p>
<p><span style="font-size: 18px;font-weight: bold;">How It
Works:</span></p>
<p><span style="font-size: 16px;"><strong><em>Importing GWAS
Results:</em></strong> Users begin by importing their GWAS results into
the geneNR package. The package includes detailed instructions and a
sample_data file to guide users through this process.</span></p>
<p><span style="font-size: 16px;"><strong><em>Identification of
Candidate Genes:</em></strong> Once the GWAS results are imported,
geneNR identifies candidate genes within the specified search window
based on the SNPs provided. This automated process not only eliminates
the need for manual searches but also significantly reduces the time
required, ensuring both accuracy and efficiency.</span></p>
<p><span style="font-size: 16px;"><strong><em>Exporting
Results:</em></strong> After the candidate genes are identified, the
package exports the results into a ready-to-use output. This format is
convenient for further analysis, reporting, and sharing with
collaborators.</span></p>
<p><span style="font-size: 18px;font-weight: bold;">Installation:</span></p>
<p><span style="font-size: 16px;">To install geneNR package, use the
following commands in R:</span></p>
<p><span style="font-size: 16px;">The geneNR package is currently in the
process of being submitted to CRAN. Once accepted, users will be able to
install the package directly from CRAN using the following command in
R:</span></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Install the geneNR package (when published on CRAN)</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;geneNR&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>or </span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co"># Installation from GitHub (if not available on CRAN)</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="co"># Uncomment and run the following commands manually:</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="co"># if (!requireNamespace(&quot;devtools&quot;, quietly = TRUE)) {</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="co">#   install.packages(&quot;devtools&quot;)</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co"># devtools::install_github(&quot;Nirmalaruban/geneNR&quot;)</span></span></code></pre></div>
<p><span style="font-size: 18px;font-weight: bold;">Note:</span></p>
<p><span style="font-size: 16px;">As the geneNR package has not been
officially published yet, researchers who wish to use it in their
studies are encouraged to contact R. Nirmalaruban (Maintainer) at <a href="mailto:nirmalaruban97@gmail.com" class="email">nirmalaruban97@gmail.com</a>. This will ensure that any
updates or issues can be addressed promptly, and users can receive the
latest information about the package.</span></p>
<p><span style="font-size: 18px;font-weight: bold;">Data:</span></p>
<p><span style="font-size: 16px;">The geneNR package includes sample
data for both wheat and rice to demonstrate its functionality. Users can
provide their GWAS result files in .csv format to mine candidate
genes.</span></p>
<p><span style="font-size: 18px;">Sample Data Skeleton:</span></p>
<p><span style="font-size: 16px;">Here is a sample data skeleton for
GWAS results:</span></p>
<p><span style="font-size: 16px;"></p>
<pre><code>| traits  |   SNP       | Chr  |   Pos   |
|----------------------------------------|
|  PH     | AX-94490431 | 7A   |93007534 |
|         |             |      |         |</code></pre>
<p></span></p>
<p><span style="font-size: 16px;">This table represents the GWAS result
data structure required by the geneNR package. Users can format their
GWAS results in a similar manner and save them as .csv files for
input.</span></p>
<p><span style="font-size: 18px;font-weight: bold;">Note:</span></p>
<p><span style="font-size: 16px;">For smooth functioning of the package,
please ensure that the data files are placed in the working
directory.</span></p>
<p><span style="font-size: 18px;font-weight: bold;">Parameters:</span></p>
<p><span style="font-size: 16px;">The function accepts the following
parameters:</span></p>
<p><span style="font-size: 16px;"></p>
<p>data_file: The input data in .csv format.</p>
<p>upstream: The search window upstream of the current position of the
SNP.</p>
<p>downstream: The search window downstream of the current position of
the SNP.</p>
<p>Crop: Either “wheat” or “rice”. </span></p>
<p><span style="font-size: 18px;font-weight: bold;">Note:</span></p>
<p><span style="font-size: 16px;">Both upstream and downstream can be
specified by the user in base pairs. By default, the search window is
set to 10^6 bp (1 Mbp).</span></p>
<p><span style="font-size: 18px;font-weight: bold;">Output:</span></p>
<p><span style="font-size: 16px;">Upon running the function, the results
will be saved in a filtered_gene_id.csv file in the current working
directory. This file will contain the following columns: traits, SNP,
search_window, gene_size, gene_id, and gene_type. The candidate genes
for the respective regions are retrieved from Ensembl Plants (<a href="https://plants.ensembl.org/index.html" class="uri">https://plants.ensembl.org/index.html</a>).</span></p>
<p><span style="font-size: 18px;">Output Format Example:</span></p>
<p><span style="font-size: 16px;"></p>
<pre><code>| traits  |  SNP  |  search_window | gene_size | gene_id | gene_type    |
|-----------------------------------------------------------------------|
|         |       |                |           |         |protein coding|</code></pre>
<p></span></p>
<p><span style="font-size: 16px;">By using the geneNR package,
researchers can significantly enhance the efficiency and accuracy of
their post-GWAS analyses, ultimately contributing to the advancement of
genomic research and crop improvement.</span></p>
<p><span style="font-size: 18px;font-weight: bold;">Glossary:</span></p>
<p><span style="font-size: 16px;"> - <strong>GWAS</strong>: Genome-Wide
Association Studies<br />
- <strong>SNP</strong>: Single Nucleotide Polymorphism<br />
- <strong>API</strong>: Application Programming Interface<br />
- <strong>Mbp</strong>: Megabase Pair (1,000,000 base pairs)<br />
- <strong>CSV</strong>: Comma-Separated Values<br />
</span></p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
